### ISSUE TITLE
Playing around with Docker Compose v3 Causes Daemon to Die

### ISSUE THREAD

**Issue by ingshtrom** [docker/for-mac#1255]
### Expected behavior
Docker still works.

### Actual behavior
I continually get the response "Error response from daemon: Bad response from Docker engine". I have to restart my machine or Docker to get rid of the error.

### Information

  - Full output of the diagnostics from "Diagnose & Feedback" in the menu
Diagnostic ID: 18B8D94B-7823-486D-90C8-6E062476EAAB
```
Docker for Mac: version: 1.13.0 (0c6d765c5)
macOS: version 10.12.3 (build: 16D32)
logs: /tmp/18B8D94B-7823-486D-90C8-6E062476EAAB/20170203-075926.tar.gz
failure: docker ps failed: (Failure "docker ps: timeout after 10.00s")
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         docker ps failed
[OK]     menubar
[OK]     disk
```

  - Docker Compose file (not that you can actually deploy these services, but maybe it's just a problem with the way I'm using Docker)
```
version: '3'

services:

  api:
    image: odyssey/olympia:master_d33ca4f8
    environment:
      - APP_ROLE=api
      - NODE_ENV=production
      - ODYSSEY_ENV=master
      - ODYSSEY_RUN=api.js
    ports:
      - 8080:8080/tcp
    networks:
      - swarmnet
    volumes:
      - ${REPOS_ROOT}/olympia:/opt/olympia
    deploy:
      replicas: 2
      update_config:
        parallelism: 1
        delay: 10s

  web:
    image: odyssey/olympia:master_d33ca4f8
    environment:
      - APP_ROLE=web
      - NODE_ENV=production
      - ODYSSEY_ENV=master
      - ODYSSEY_RUN=server.js
    ports:
      - 8070:8000/tcp
    networks:
      - swarmnet
    depends_on:
      - api
    volumes:
      - ${REPOS_ROOT}/olympia:/opt/olympia
    deploy:
      replicas: 2
      update_config:
        parallelism: 1
        delay: 10s

  kong-database:
    image: postgres:9.4
    networks:
      - swarmnet
    environment:
      - POSTGRES_USER=kong
      - POSTGRES_DB=kong
    ports:
      - 5432:5432/tcp

  kong:
    image: odyssey/kong:production_554688cc
    networks:
      - swarmnet
    ports:
      - 8001:8001/tcp
      - 8443:8443/tcp
      - 8000:8000/tcp
      - 7492:7492/udp
    environment:
      - KONG_DATABASE=postgres
      - KONG_CASSANDRA_CONTACT_POINTS=kong-database
      - KONG_PG_HOST=kong-database
      - KONG_LOG_LEVEL=info
    depends_on:
      - kong-database
    deploy:
      replicas: 1
      update_config:
        parallelism: 1
        delay: 30s

  nginx:
    image: odyssey/nginx:production_d20a01ce
    networks:
      - swarmnet
    ports:
      - 80:80/tcp
      - 443:443/tcp
    depends_on:
      - api
      - web
      - kong
    deploy:
      mode: global
      update_config:
        parallelism: 1
        delay: 10s

networks:
  swarmnet:
    external: true
```

### Steps to reproduce the behavior
I cannot reproduce it every time, but every now and then, after running `docker stack deploy -c default-stack.yaml test`, it fails. 

I *feel* like I saw these problems more after I added the `volume` properties in for the `web` and `api` services, but now after I take them out and put them back in, I can't reproduce the problem. I did notice that I have an invalid mount source, so maybe it got in a restart loop and caused Docker daemon to crash?

**Comment by samoht** [comment]
There seems that something goes indeed wrong with non-existent volumes:

```
Feb  3 12:58:30 moby root: time="2017-02-03T12:58:30.117022997Z" level=error msg="failed to start taskManager" error="invalid bind mount source, source path not found: /olympia" module="node/agent"
Feb  3 12:58:30 moby root: panic: runtime error: invalid memory address or nil pointer dereference
Feb  3 12:58:30 moby root: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xfc0854]
Feb  3 12:58:30 moby root: goroutine 39774 [running]:
Feb  3 12:58:30 moby root: panic(0x176af20, 0xc4200120a0)
Feb  3 12:58:30 moby root: ^I/usr/local/go/src/runtime/panic.go:500 +0x1a1
Feb  3 12:58:30 moby root: github.com/docker/docker/vendor/github.com/docker/swarmkit/agent.(*taskManager).Logs(0x0, 0x250f8e0, 0xc422b3eff0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x24fc4e0, ...)
Feb  3 12:58:30 moby root: ^I/go/src/github.com/docker/docker/vendor/github.com/docker/swarmkit/agent/task.go:68 +0xa4
Feb  3 12:58:30 moby root: created by github.com/docker/docker/vendor/github.com/docker/swarmkit/agent.(*worker).Subscribe
Feb  3 12:58:30 moby root: ^I/go/src/github.com/docker/docker/vendor/github.com/docker/swarmkit/agent/worker.go:511 +0x54c
Feb  3 12:58:30 moby root: panic: runtime error: invalid memory address or nil pointer dereference
Feb  3 12:58:30 moby root: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xfc0854]
Feb  3 12:58:30 moby root: goroutine 39773 [running]:
Feb  3 12:58:30 moby root: panic(0x176af20, 0xc4200120a0)
Feb  3 12:58:30 moby root: ^I/usr/local/go/src/runtime/panic.go:500 +0x1a1
Feb  3 12:58:30 moby root: github.com/docker/docker/vendor/github.com/docker/swarmkit/agent.(*taskManager).Logs(0x0, 0x250f8e0, 0xc421884720, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x24fc4e0, ...)
Feb  3 12:58:30 moby root: ^I/go/src/github.com/docker/docker/vendor/github.com/docker/swarmkit/agent/task.go:68 +0xa4
Feb  3 12:58:30 moby root: created by github.com/docker/docker/vendor/github.com/docker/swarmkit/agent.(*worker).Subscribe
Feb  3 12:58:30 moby root: ^I/go/src/github.com/docker/docker/vendor/github.com/docker/swarmkit/agent/worker.go:511 +0x54c
```

I will report the issue upstream. Thanks for your report!

**Comment by friism** [comment]
This was reportedly fixed in Docker 1.13.1, closing. Please re-open if you're still seeing this problem.

https://github.com/docker/docker/issues/30711
